docs/articles/documentation/recipes/integrating-testcafe-with-ci-systems/gitlab.md

```diff
@@ -0,0 +1,111 @@
+
+The easiest way to run TestCafe tests is [use an official TestCafe Docker image](#option-1---use-testcafe-docker-image) based on Alpine Linux with preinstalled Node.js, TestCafe, Firefox and Chromium.
+
+However, if you already have a Docker image prepared to deploy your web application and run tests, you can [install TestCafe to this image before testing](#option-2---install-testcafe-to-a-docker-image).
+
+## Option 1 - Use TestCafe Docker Image
+
+Open the `.gitlab-ci.yml` file that stores GitLab CI configuration. If you don't have this file yet, create it in the repository's root directory.
+
+Create a new [job](https://docs.gitlab.com/ee/ci/pipelines.html#jobs). To do this, add the following settings to `.gitlab-ci.yml` at the top level:
+
+```yaml
+e2e_tests:
+  image:
+    name: testcafe/testcafe
+    entrypoint: ["/bin/sh", "-c"]
+  script:
+    - /opt/testcafe/docker/testcafe-docker.sh firefox:headless,chromium tests/**/*
+```
+
+* `e2e_tests` - the job's name. You can choose any name you wish.
+* `image` - Docker image settings.
+* `name` - the TestCafe image's name. You can find it on [Docker Hub](https://hub.docker.com/r/testcafe/testcafe/).
+* `entrypoint` - overrides the image's [ENTRYPOINT](https://docs.docker.com/glossary/?term=ENTRYPOINT) to be `/bin/sh`. You need to specify this setting because the TestCafe image has the default `ENTRYPOINT` set to the `testcafe` command, while GitLab uses `sh` to run test scripts. The `-c` flag indicates that the subsequent argument is a command rather than a script.
+* `script` - the command that runs TestCafe tests. `/opt/testcafe/docker/testcafe-docker.sh` points to a script that prepares the environment to run a browser and starts TestCafe. Its arguments are standard TestCafe [command line parameters](../../using-testcafe/command-line-interface.md).
+
+For more information about the `.gitlab-ci.yml` syntax, see [Configuration of your jobs with .gitlab-ci.yml](https://docs.gitlab.com/ee/ci/yaml/README.html).
+
+Commit `.gitlab-ci.yml` and push the changes to the repository. Now you have a new [job](https://docs.gitlab.com/ee/ci/pipelines.html#jobs) that can be executed by any [GitLab Runner](https://docs.gitlab.com/ee/ci/runners/README.html).
+
+## Option 2 - Install TestCafe to a Docker Image
+
+If you use a Docker image that does not have TestCafe preinstalled, you can install it from `npm` before tests are run.
+
+First, add TestCafe to your project development dependencies. Open the repository root and execute the following command:
+
+```sh
+npm --save-dev testcafe
+```
+
+This installs the latest TestCafe version locally and adds it to the `devDependencies` section in the `package.json` file.
+
+```json
+{
+    "devDependencies": {
+        "testcafe": "*"
+    }
+}
+```
+
+Since TestCafe is installed locally, the test run command that uses TestCafe should be also added to `package.json` to the `scripts` section.
```

Comment:
Since TestCafe is installed locally, the test run command that uses TestCafe should be also added to `package.json` to the `scripts` section -> The test run command that uses TestCafe should be also added to `package.json` to the `scripts` section because TestCafe is installed locally.